content/en/blog/_posts/2023-08-21-job-update-post.md

```diff
@@ -0,0 +1,207 @@
+
+This is an alpha feature, which you can enable by turning on the
+`JobBackoffLimitPerIndex`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
+in your cluster.
+
+Once the feature is enabled, you can create an Indexed Job with the
+`.spec.backoffLimitPerIndex` field specified.
+
+#### Example
+
+The following example demonstrates how to use this feature to make sure the
+Job executes all indexes, and the number of failures is controller per index.
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: job-backoff-limit-per-index-execute-all
+spec:
+  completions: 8
+  parallelism: 2
+  completionMode: Indexed
+  backoffLimitPerIndex: 1
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: example
+        image: python
+        command:
+        - python3
+        - -c
+        - |
+          import os, sys, time
+          id = int(os.environ.get("JOB_COMPLETION_INDEX"))
+          if id == 1 or id == 2:
+            sys.exit(1)
+          time.sleep(1)
+```
+
+Now, inspect the pods after the job is finished:
+
+```sh
+kubectl get pods -l job-name=job-backoff-limit-per-index-execute-all
+```
+
+Returns output similar to this:
+```
+NAME                                              READY   STATUS      RESTARTS   AGE
+job-backoff-limit-per-index-execute-all-0-b26vc   0/1     Completed   0          49s
+job-backoff-limit-per-index-execute-all-1-6j5gd   0/1     Error       0          49s
+job-backoff-limit-per-index-execute-all-1-6wd82   0/1     Error       0          37s
+job-backoff-limit-per-index-execute-all-2-c66hg   0/1     Error       0          32s
+job-backoff-limit-per-index-execute-all-2-nf982   0/1     Error       0          43s
+job-backoff-limit-per-index-execute-all-3-cxmhf   0/1     Completed   0          33s
+job-backoff-limit-per-index-execute-all-4-9q6kq   0/1     Completed   0          28s
+job-backoff-limit-per-index-execute-all-5-z9hqf   0/1     Completed   0          28s
+job-backoff-limit-per-index-execute-all-6-tbkr8   0/1     Completed   0          23s
+job-backoff-limit-per-index-execute-all-7-hxjsq   0/1     Completed   0          22s
+```
+
+Additionally, let's take a look at the job status:
+
+```sh
+kubectl get jobs job-backoff-limit-per-index-fail-index -o yaml
+```
+
+Returns output similar to this:
+
+```yaml
+  status:
+    completedIndexes: 0,3-7
+    failedIndexes: 1,2
+    succeeded: 6
+    failed: 4
+    conditions:
+    - message: Job has failed indexes
+      reason: FailedIndexes
+      status: "True"
+      type: Failed
+```
+
+Here, indexes `1`  and `2` were both retried once. After the second failure,
+in each of them, the specified `.spec.backoffLimitPerIndex` was exceeded, so
+the retries were stopped. For comparison, if the per-index backoff was disabled,
+then the buggy indexes would retry until the global `backoffLimit` was exceeded,
+and then the entire Job would be marked failed, before some of the higher
+indexes are started.
+
```

Comment:
@mimowo Maybe we should link to the KEP for people to learn more?  As I did above.  

It would probably be nice to link to the KEP for interested readers.

Reply:
Yeah, added in the section that is now moved to the bottom.